src/core/manager.c

```diff
@@ -684,8 +681,11 @@ int manager_default_environment(Manager *m) {
                 if (!m->transient_environment)
                         return log_oom();
 
-                r = strv_env_replace_strdup(&m->transient_environment, "PATH=" DEFAULT_USER_PATH);
-                if (r < 0)
+                _cleanup_free_ char *path = NULL;
+                if (asprintf(&path, "PATH=%s", default_user_path()) < 0)
```

Comment:
please do not concat paths with asprintf(), way too expensive for that. 

please use strjoin()